docs/telepathy-rakia.8.in
.TH TELEPATHY-RAKIA "8" "February 2010" "Telepathy" "D-Bus services"
\" This man page was written by Simon McVittie for the Debian project,
\" but may be used by others.
\" Copyright © 2007-2010 Collabora Ltd. <http://www.collabora.co.uk/>
\" It may be distributed under the same terms as telepathy-rakia itself.
.SH NAME
telepathy-rakia \- Telepathy connection manager for SIP
.SH SYNOPSIS
\fB@libexecdir@/telepathy\-rakia\fR
.SH DESCRIPTION
\fBtelepathy\-rakia\fR implements the Telepathy D-Bus specification for
the Session Initiation Protocol, allowing Telepathy clients like
.BR empathy (1)
to connect to Voice over IP providers that implement SIP.
.PP
It is a D-Bus service which runs on the session bus, and should usually be
started automatically by D-Bus activation. However, it might be useful to
start it manually for debugging.
.SH OPTIONS
There are no command-line options.
.SH ENVIRONMENT
.TP
\fBRAKIA_DEBUG\fR=\fItype\fR
May be set to "all" for full debug output, or various undocumented options
(which may change from release to release) to filter the output.
.TP
\fBRAKIA_LOGFILE\fR=\fIfilename\fR
May be set to the name of a file to overwrite with log messages. The filename
may be prefixed with \fB+\fR to append messages to an existing file instead
of overwriting it.
.TP
\fBRAKIA_PERSIST\fR
May be set to any value to avoid telepathy-rakia's usual automatic exit
when there have been no connections for a few seconds.
.TP
\fBTPORT_LOG\fR
May be set to any value to print all parsed SIP messages at the transport
layer (this functionality is provided by the underlying Sofia-SIP library,
which also offers various more specialized debugging options).
.SH SEE ALSO
.IR http://telepathy.freedesktop.org/ ,
.BR empathy (1)


NEWS
===========================
NEWS for telepathy-sofiasip
===========================

List of changes in each public release of telepathy-sofiasip (most 
recent at the top).

See also ChangeLog.

Changes in release 0.6.8 (2011-06-14)
-------------------------------------

The "homage to World's Toughest Fixes" release.

- Refactored handling of local codec information to avoid excessive re-INVITEs
- Remove non-intersecting local codecs from any further SDP
  negotiations (fd.o #30434).


Changes in release 0.6.7 (2011-05-23)
-------------------------------------

The "boarding up" release.

- Verify the validity of TLS certificates presented by SIP connection peers.
  This change is disruptive: it relies on root CA certificates being available
  to sofia-sip in the default verification path ``~/.sip/auth`` or file
  ``~/.sip/auth/cafile.pem``, or sofia-sip changed to use OpenSSL library
  defaults for verification path (sf.net #3306245).
  The connection parameter "ignore-tls-errors" is added to disable
  verification.
- Choose keepalive timing to make the iphb heartbeat synchronize with the
  other applications. (fd.o #37354)
- Omission of parameter 'keepalive-interval' does not mean disabled keepalives.
  A default defined by sofia-sip is used instead.
- More client-friendly behavior for connection parameter 'discover-stun':
  if set to TRUE, overrides 'stun-server'; by default discovery is performed if
  'stun-server' is not set. (fd.o #34227)
- If the home domain is given as an IP address, don't try to perform
  SRV lookups on it.
- Requires sofia-sip >= 1.12.11.
- Requires iphb >= 0.61.31, if used.

Changes in release 0.6.6 (2011-02-01)
-------------------------------------

The "Multiple Personality Communications Disorder" release.

- Reject incoming calls from the same URI as the local user, rather than
  crashing apologetically (fd.o #33716).

Changes in release 0.6.5 (2011-01-29)
-------------------------------------

The "make buildable" release.

- Changed the type of the size parameter to gssize in tpsip_unquote_string(). 
- Bugs fixed:
  - fd.o #31720 - stop using tp_get_bus()

Changes in release 0.6.4 (2010-08-27)
-------------------------------------

The "stretched thin" release.

- Fix a thinko preventing proper declining of incoming calls with
  informative statuses like 603 Decline.

Changes in release 0.6.3 (2010-06-21)
-------------------------------------

The "my attorney suggests I do this" release.

- Disable Sofia logging on --disable-debug configure flag.
- Made the effect of environment variable TPSIP_PERSIST independent of
  the debug build flag.
- Prevent removal of local pending send by holding the call.
- Fix the application of local pending send flag in an incoming call.
- Fixed a walk-off-the-end crash in the debug support code.
- Not building useless PIC object files any more.

Changes in release 0.6.2 (2010-03-17)
-------------------------------------

The "I can't hear you breathing nervously" release.

- Only expose the legacy STUN properties when set to meaningful values.
- Properly enable the debug category for Sofia-SIP log messages.
- Fix a crash in Sofia logger (fd.o #26889).
- Fixed a regression that leaked audio before the call was accepted.
- On incoming call, emit the new channel with correct initial media properties.

Changes in release 0.6.1 (2010-02-19)
-------------------------------------

The "people who stare at the ceiling" release.

- Avoid using IPv6 addresses in self-referential URIs (fd.o #26654)
- Provide a NEWS file with highlights of previous releases

Changes in release 0.6.0 (2010-02-15)
-------------------------------------

- Supports the new StreamHandler interface method CodecsUpdated.
- Revisited code controlling stream directionality and pending flags
  (fd.o #26070).
- Made critical warnings fatal again, after telepathy-glib stopped doing so.
  (fd.o #26153).
- Try to disable locally originated re-INVITES as much as possible
  when 'immutable-streams' parameter is true (fd.o #26387).
- Implemented the Debug interface (fd.o #26388).
- Implemented the DBus properties of Media.StreamHandler (fd.o #26384).
- Build fixes and improvements.
- Support configure option --enable-silent-rules if using Automake >= 1.11
- Implemented properties InitialAudio, InitialVideo, and ImmutableStreams.
- Introduced the connection manager parameter 'immutable-streams' to
  disable stream list changes mid-call, as well as hold.
- Pass the ptime and maxptime attributes from remote SDP session and media
  descriptions as codec parameters. Values from a=fmtp are also considered. 

Changes in release 0.5.19 (2009-12-02)
--------------------------------------

- Implemented proper codec parameter formatting for telephone/event
  payloads, and the supporting infrastructure for more formats.
  (fd.o bug #24615)
- Ignore the trivial alias for self handle that is produced from the
  SIP URI.
- Implemented call state In_Progress.
- Properly flag password parameters as secret, also in the .manager file
  (fd.o bug #24561).
- Autotools cleanups.
- Normalize the alias for self handle, and properly format it into a
  double-quoted string for the SIP From header.
- Use heartbeat more carefully, with per-connection wakeups timed
  in accordance with actual keepalive intervals.
- Eliminate race conditions in hold/unhold code.

Changes in release 0.5.18 (2009-08-27)
--------------------------------------

- Don't delay the final 200 response to incoming messages.
- Use sips: scheme in URI normalization when TLS transport
  is selected.
- Treat 183 Session Progress response with setting the Ringing call state.

Changes in release 0.5.17 (2009-07-25)
--------------------------------------

- Do not check an unwarranted handle assertion (fd.o bug #17751).
- Optionally using iphb on Maemo as a trigger for keepalives.

Changes in release 0.5.16 (2009-04-14)
--------------------------------------

- Updated to Telepathy specification 0.17.22:

src/telepathy-rakia.c
/*
 * sip-connection.c - Source for RakiaConnection
 * Copyright (C) 2005 Collabora Ltd.
 * Copyright (C) 2005-2008, 2010 Nokia Corporation
 *   @author Kai Vehmanen <first.surname@nokia.com>
 *   @author Martti Mela <first.surname@nokia.com>
 *   @author Mikhail Zabaluev <mikhail.zabaluev@nokia.com>
 *
 * Based on telepathy-gabble implementation (gabble-connection).
 *   @author See gabble.c
 *
 * This work is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 2.1 of the License, or (at your option) any later version.
 *
 * This work is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public
 * License along with this work; if not, write to the Free Software
 * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 */

#include "config.h"

#include "rakia/debug.h"

#include "sip-connection-manager.h"
#include <telepathy-glib/run.h>
#include <telepathy-glib/debug.h>

static TpBaseConnectionManager *
construct_cm (void)
{
  return (TpBaseConnectionManager *)g_object_new (
      RAKIA_TYPE_CONNECTION_MANAGER, NULL);
}


int
main (int argc, char** argv)
{
  int status;
  gpointer logdata;
  guint fatal_mask;

  fatal_mask = g_log_set_always_fatal (G_LOG_FATAL_MASK);
  fatal_mask |= G_LOG_LEVEL_CRITICAL;
  g_log_set_always_fatal (fatal_mask);  

#ifdef ENABLE_DEBUG
  rakia_debug_set_flags_from_env ();
#endif

  if (g_getenv ("RAKIA_PERSIST") || g_getenv ("RAKIA_PERSIST"))
    {
      tp_debug_set_persistent (TRUE);
    }

  tp_debug_divert_messages (g_getenv ("RAKIA_LOGFILE"));

  logdata = rakia_sofia_log_init ();

  status = tp_run_connection_manager ("telepathy-rakia", VERSION,
                                      construct_cm, argc, argv);

  rakia_sofia_log_finalize (logdata);

  return status;
}
